src/alfred/ctx.py

```diff
@@ -109,18 +113,24 @@ def mode_set(mode: str) -> None:
     _invocation_context.mode = mode
 
 
-def invoke_through_external_venv(args: List[str]) -> None:
+def invoke_through_external_venv(args: List[str], pty: bool = False) -> None:
     """
     Invokes the current alfred command in its own virtual environment.
 
     """
     alfred_cmd = current_command()
     venv = interpreter.venv_lookup(alfred_cmd.project_dir)
-    exit_code, _, stderr = interpreter.run_module(module='alfred.cli', venv=venv, args=args)
+    if pty is True:
+        try:
+            interpreter.run_module_as_pty(module='alfred.cli', venv=venv, args=args)
```

Comment:
It would be better to have a prototype close to ``interpreter.run_module`` and do the verification on the exit_code.